File: managingActivity LevelPreferences/app/src/main/java/com/example/managingactivitylevelpreferences/MainActivity.java
package com.example.managingactivitylevelpreferences;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.Switch;

public class MainActivity extends AppCompatActivity {

    Switch mswitch;
    View parentView;
    final String COLOR_KEY = "color";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mswitch = findViewById(R.id.switch1);
        parentView = findViewById(R.id.parentLayout);

        SharedPreferences Preferences = getPreferences(MODE_PRIVATE);
        boolean isChecked = Preferences.getBoolean(COLOR_KEY,false);

        mswitch.setChecked(isChecked);

        parentView.setBackgroundColor(isChecked?Color.GREEN:Color.YELLOW);

        mswitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {

                //note that we have not written getSharedPreferences because it is used when you are working for app and here working for activity so used this
                //means getPreferences note if you have accessed it from outside the this activity then you can not retrive it.
                SharedPreferences sharedPreferences = getPreferences(MODE_PRIVATE);
                SharedPreferences.Editor editor= sharedPreferences.edit();

                editor.putBoolean(COLOR_KEY,isChecked);
                editor.apply();
                parentView.setBackgroundColor(isChecked? Color.GREEN : Color.YELLOW);
            }
        });


    }
}

File: savedSharedPreference/app/src/main/java/com/example/savedsharedpreference/MainActivity.kt
package com.example.savedsharedpreference

import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    lateinit var textView: TextView
    lateinit var editText: EditText
    lateinit var saveButton: Button
    lateinit var applyTextButton: Button
    private var switch1: Switch? = null
    private var text: String? = null
    private var switchstatus = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        textView = findViewById(R.id.textview)
        editText = findViewById(R.id.edittext)
        saveButton = findViewById(R.id.save_button)
        applyTextButton = findViewById(R.id.apply_text_button)
        switch1 = findViewById(R.id.switch1)
        applyTextButton.setOnClickListener(View.OnClickListener {
            textView.setText(editText.getText().toString())
        })
        saveButton.setOnClickListener(View.OnClickListener { saveData() })
        loadData()
        updateData()
    }

    fun saveData() {
        val sharedPreferences = getSharedPreferences("saved", MODE_PRIVATE)
        val editor = sharedPreferences.edit()
        editor.putString("Value", textView!!.text.toString())
        editor.putString("SWITCH1", switch1!!.isChecked.toString())
        editor.apply()
        Toast.makeText(this, "Data is saved", Toast.LENGTH_SHORT).show()
    }

    fun loadData() {
        val sharedPreferences = getSharedPreferences("saved", MODE_PRIVATE)
        text = sharedPreferences.getString("Value", "")
        switchstatus = sharedPreferences.getBoolean(switch1.toString(), false)
    }

    fun updateData() {
        textView!!.text = text
        switch1!!.isChecked = switchstatus
    }
}